Serializer/Serializeables/NSDictionary+Serializable.m

```
@@ -0,0 +1,56 @@
+#import "NSDictionary+Serializable.h"
+#import "Serializer.h"
+#import "Constants.h"
+
+@implementation NSDictionary (Serializable)
+
+- (NSString *)serializeWithError:(NSError **)error
+{
+	NSMutableString *result = [[NSMutableString alloc] initWithFormat:@"dictionary: {"];
+
+	for (id<Serializable> key in self)
+	{
+		if ([key isKindOfClass:[NSString class]] || [key isKindOfClass:[NSNumber class]])
+		{
+			NSString *serializedKey = [Serializer serialize:key withError:error];
+
+			if (serializedKey != nil)
+			{
+				NSString *serializedValue = [Serializer serialize:self[key] withError:error];
+
+				if (serializedValue != nil)
+				{
+					[result appendFormat:@"%@: %@,\n", serializedKey, serializedValue];
+				}
+				else
+				{
+					return nil;
+				}
+			}
+			else
+			{
+				return nil;
```

Why not return an error here?